.github/actions.py

```diff
@@ -85,14 +99,32 @@ def update(pkg_name, version, link):
     index_file = os.path.join(norm_pkg_name, INDEX_FILE) 
     with open(index_file) as html_file:
         soup = BeautifulSoup(html_file, "html.parser")
+        
+    # Extract the URL from the onclick attribute
+    button = soup.find('a', id='repoHomepage')
+    if button:
+        link = button.get("href")
+    else:
+        raise Exception("Homepage URL not found")
 
     # Create a new anchor element for our new version
-    last_anchor = soup.find_all('a')[-1]        # Copy the last anchor element
-    new_anchor = copy.copy(last_anchor)
-    new_anchor['href'] = "{}#egg={}-{}".format(link, norm_pkg_name, version)
+    original_div = soup.find('section', class_='versions').findAll('div')[-1]
+    new_div = copy.copy(original_div)
+    anchors = new_div.find_all('a')
+    new_div['onclick'] = "load_readme('{}', scroll_to_div=true)".format(version)
+    new_div['id'] = version
+    new_div['class'] = ""
+    if 'dev' in version:
+        new_div['class'] += "prerelease"
+    else:
+        # replace the latest main version
+        main_version_span = soup.find('span', id='latest-main-version')
+        main_version_span.string = version
+    anchors[0].string = version
+    anchors[1]['href'] = "git+{}@{}#egg={}-{}".format(link,version,norm_pkg_name,version)
```

Comment:
We should have the same in the delete method : if the latest stable version is removed, we should set the previous version as latest stable.

Reply:
Oh actually nevermind : even if we remove that version from the PyPi index, it's still the latest stable in the github repository.

So we shouldn't mark the previous version as the latest stable !

Reply:
The delete method deletes all the package, it doesn't delete versions

Reply:
Right xD
I forgot 💀